azkaban-common/src/main/java/azkaban/executor/dispatcher/Dispatcher.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2015 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.executor.dispatcher;
+
+import java.util.List;
+
+
+/** Definition of the dispatcher interface.
+ *  an implementation of the dispatcher interface provides the functionality
+ *  to return a candidate from the candidateList that suits best for the dispatchingObject.
+ *  @param K : type of the candidate.
+ *  @param V : type of the dispatching object.
+ */
+public interface Dispatcher <K,V> {
```

Comment:
true,  the interface only selects a candidate to handle the work but does not actually put the task on to it . 

please see a design pattern discussion thread below, in which the dispatcher classes does the similar @ http://www.javaworld.com/article/2075719/design-patterns/dispatcher-eases-workflow-implementation.html